core/src/main/scala/cats/TraverseFilter.scala

```diff
@@ -69,7 +68,7 @@ trait TraverseFilter[F[_]] extends FunctorFilter[F] {
    * res0: List[Option[String]] = List(Some(two), None)
    */
   def traverseCollect[G[_], A, B](fa: F[A])(f: PartialFunction[A, G[B]])(implicit G: Applicative[G]): G[F[B]] =
-    traverseFilter(fa)(a => f.lift(a).sequence)
+    traverseFilter(fa)(a => f.lift(a).map(G.map(_)(Option(_))).getOrElse(G.pure(None)))
```

Comment:
I think `sequence` was good, we just can't use the syntax :)

```suggestion
    traverseFilter(fa)(a => Traverse[Option].sequence(f.lift(a)))
```

Reply:
actually, while we are at it:

```scala
val optF = f.lift
traverseFilter(fa)(a => Traverse[Option].sequence(optF(a)))
```
so we don't reallocate `f.lift` on every item `a`.

Reply:
@emilhotkowski I think Oscar's suggestion still applies :) it's a good optimization.

Reply:
Added!